packages/compass-indexes/src/components/create-index-fields/create-index-fields.tsx

```diff
@@ -0,0 +1,260 @@
+import React, { Component } from 'react';
+import type { ReactElement } from 'react';
+import { hasColumnstoreIndexesSupport } from '../../utils/has-columnstore-indexes-support';
+
+import {
+  Combobox,
+  ComboboxOption,
+  Select,
+  Option,
+  IconButton,
+  spacing,
+  css,
+  Icon,
+} from '@mongodb-js/compass-components';
+
+import type { IndexField } from '../../modules/create-index/fields';
+
+// Inject types that the ComboboxOption component accepts.
+// Otherwise, ComboboxOption causes
+// the `Property 'value' does not exist on type 'IntrinsicAttributes'` error.
+const ComboboxOptionTyped =
+  ComboboxOption as unknown as React.JSXElementConstructor<{
+    value?: string;
+    displayName?: string;
+    glyph?: ReactElement;
+    className?: string;
+  }>;
+
+/**
+ * Current allowed types for indexes.
+ */
+const INDEX_TYPES = ['1 (asc)', '-1 (desc)', '2dsphere', 'text', 'columnstore'];
+
+/**
+ * Default values for field name and type as presented in the UI.
+ */
+const DEFAULT_FIELD = {
+  name: 'Select or type a field name',
+  type: 'Select a type',
+};
+
+const hostActionButtonStyles = css({
+  marginLeft: spacing[1],
+  marginTop: spacing[1],
+});
+
+const createIndexFieldsStyles = css({
+  display: 'flex',
+  justifyContent: 'stretch',
+  marginBottom: spacing[2],
+  marginTop: spacing[1],
+});
+
+const createIndexFieldsNameStyles = css({
+  width: +spacing[7] * 3 + +spacing[6],
```

Comment:
Do we need this width and the width below on `createIndexFieldsTypeStyles`? It seems to look alright without it. Asking because this calculation looks like something that would have to update if we update `spacing` and if we can avoid that dependency it might be nice. Fine as is too.

Reply:
If do not set width explicitly it changes the size of components when name or type is selected. I changed it `vw` tho so it does not depend on spacing.

Reply:
Gotcha, in that case I think `spacing` might be better here then `vw`, with `vw` the width of these elements could be huge on large/stretched monitors.

Reply:
Reverted to spaces :)